packages/wallets/src/AppContent.tsx

```diff
@@ -9,6 +9,11 @@ const AppContent: React.FC = () => {
     const [isPanelOpen, setIsPanelOpen] = useState(false);
     const { i18n } = useTranslation();
 
+    useLayoutEffect(() => {
+        const evt = new Event('walletsIsMounting');
+        document.dispatchEvent(evt);
+    }, []);
+
```

Comment:
Here we are dispatching an event in order to intercept it in `Core` package to switch the theme to light mode, if the user is trying to reach `wallets` route with previously enabled dark mode.
We are doing this `magic` because there are no synchronization between core and wallets package.
Pls feel free to suggest other solutions 🙏

Reply:
been thinking about some explicit singleton, event-emitter to share such data, 

so then, when we want to get wallets out of core, all the core connection are in one place (instead of being spread across codebase). While still not ideal, at least keeps the problematic bits in one place and keeps them listed and documented :)

any thoughts @markwylde-deriv ?

Reply:
Are we only doing this to [disable dark mode](https://github.com/binary-com/deriv-app/blob/3b70823bd944ee147fecad3bd326c81e6d066d8f/packages/core/src/App/AppContent.tsx#L35)? I hate everything about this 😆, mostly because it requires us to do weird things in wallets. Is there other logic we think this will be used for?

When I look at the [AppContent](https://github.com/binary-com/deriv-app/blob/master/packages/core/src/App/AppContent.tsx#L74), I've thought several times how hard would it be to flip the router so it's on top. Then we could pass in the dark/light theme into the header based on the route.

Reply:
yes @markwylde-deriv 
since theme and language settings belong to core (header and footer) and wallets belong to wallets, we are doing such kind of magic. Not sure regarding our further actions with header and footer, but because now it's quite hard to sync core with wallets we are doing this

Reply:
I feel:
1. If the user is logged in, language and theme should be stored on the API
2. I'm not entirely against putting a cookie for language and theme on the `deriv.com` domain. I'm less okay with the auth token.

Reply:
feels like options are:
1) just go with current implementation
b) improve it slightly to get all the comms between core and wallest through one class/one singleton (so at least they'll be in control)
3) recognise that wallets are being used within the core itself (so that the wallets do not need to send events to core about themselves)

Reply:
ok guys so eventually i'm checking in core that we have wallet accounts and if so we intentionally switching the user to light mode